crates/biome_css_semantic/src/semantic_model/specificity.rs

```diff
@@ -0,0 +1,180 @@
+use crate::semantic_model::model::Specificity;
+
+use biome_css_syntax::{
+    AnyCssCompoundSelector, AnyCssPseudoClass, AnyCssRelativeSelector, AnyCssSelector,
+    AnyCssSimpleSelector, AnyCssSubSelector, CssComplexSelector, CssCompoundSelector,
+    CssPseudoClassSelector,
+};
+
+const ID_SPECIFICITY: Specificity = Specificity(1, 0, 0);
+const CLASS_SPECIFICITY: Specificity = Specificity(0, 1, 0);
+const TYPE_SPECIFICITY: Specificity = Specificity(0, 0, 1);
+const ZERO_SPECIFICITY: Specificity = Specificity(0, 0, 0);
+
+fn evaluate_any_simple_selector(selector: &AnyCssSimpleSelector) -> Specificity {
+    match selector {
+        AnyCssSimpleSelector::CssTypeSelector(_) => TYPE_SPECIFICITY,
+        AnyCssSimpleSelector::CssUniversalSelector(_) => ZERO_SPECIFICITY,
+    }
+}
+
+/// See https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#the_is_not_has_and_css_nesting_exceptions
+fn evaluate_pseudo_function_selector(name: &str) -> Option<Specificity> {
```

Comment:
```suggestion
const fn evaluate_pseudo_function_selector(name: &str) -> Option<Specificity> {
```

Reply:
Change to `const fn` at https://github.com/biomejs/biome/pull/4097/commits/3dbb9ca04c2ca81759db8ba161920dd1ff5f4356